src/osd/OSD.cc

```diff
@@ -1388,21 +1388,13 @@ MOSDMap *OSDService::build_incremental_map_msg(epoch_t since, epoch_t to,
   if (since < m->cluster_osdmap_trim_lower_bound) {
     // we don't have the next map the target wants, so start with a
     // full map.
-    bufferlist bl;
     dout(10) << __func__ << " cluster osdmap lower bound "
              << sblock.cluster_osdmap_trim_lower_bound
              << " > since " << since << ", starting with full map"
              << dendl;
     since = m->cluster_osdmap_trim_lower_bound;
-    if (!get_map_bl(since, bl)) {
-      derr << __func__ << " missing full map " << since << dendl;
-      goto panic;
-    }
-    max--;
-    max_bytes -= bl.length();
-    m->maps[since] = std::move(bl);
```

Comment:
This removal is wrong.  The below loop populates the map message with incremental maps.  This branch specifically handles the case where the receiving OSD will have a map gap.  In that case, the first map we send must be a full map.